ofono/bridge.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import os
+from time import sleep
+
+import pydbus
+from dbus.mainloop.glib import DBusGMainLoop
+from gi.repository import GLib
+
+from helpers import Singleton, setup_logger
+
+DBusGMainLoop(set_as_default=True)  # has to be called first
+
+logger = setup_logger(__name__, 'debug')
+logging.basicConfig(level=logging.DEBUG)
+
+
+class OfonoBridge(object):
+    """
+    Generic util class to bridge between ZPUI and ofono backend through D-Bus
+    """
+
+    def __init__(self):
+        super(OfonoBridge, self).__init__()
+
+        bus = pydbus.SystemBus()
+        manager = bus.get('org.ofono', '/')
+        modem_path = manager.GetModems()[0][0]
+
+        if modem_path != '/sim900_0':
+            raise ValueError("Default modem should be '/sim900_0', was '{}'".format(modem_path))
+
+        # self.start()  #todo: check if it's the suitable place to do it
+
+    def start(self):
+        self.power_on()
+        self._init_messages()
+        self._listen_messages()
+
+    @property
+    def _bus(self):
+        # lil hack so we always have an up-to-date bus
+        return pydbus.SystemBus().get('org.ofono', '/sim900_0')
+
+    @property
+    def message_manager(self):
+        return self._get_dbus_interface('MessageManager')
+
+    def _get_dbus_interface(self, name):
+        ''.startswith('org.ofono')
+        full_name = name if name.startswith('org.ofono') else 'org.ofono.{}'.format(name)
+        if full_name in self._bus.GetProperties()['Interfaces']:
+            return self._bus[full_name]
+
+    def power_on(self):
+        if self._bus.GetProperties()["Powered"]:
+            logger.info("Modem already powered up !")
+        else:
+            logger.info("Powering up modem...")
+            try:
+                self._bus.SetProperty("Powered", pydbus.Variant('b', True))
+                sleep(2)  # Let the modem some time to initialize
+            except Exception:
```

Comment:
1. Just `except:` is enough if you want a catch-all
2. I think that, if the modem can't power on, we'll want some more information - as much as the exception can provide us, and maybe more, so don't skimp on logging here =)

Reply:
sure